content/documentation/architecture.adoc

```diff
@@ -0,0 +1,118 @@
+---
+title: "Architecture"
+date: 2018-09-03T14:39:48-05:00
+draft: false
+---
+
+= Kiali architecture
+:imagesdir: /images/documentation/architecture
+
+Kiali is composed of two components: a backend running in the container
+application platform, and a user frontend. Moreover, Kiali depends on external
+services and components provided by other applications.
+
+The following diagram illustrates the components involved in Kiali and its
+interactions:
+
+image::architecture.png[Kiali architecture]
+
+== Kiali backend
+
+The backend is the application that runs in the container application platform.
+It's written in http://golang.org/[Go]. The code can be found at
+https://github.com/kiali/kiali[kiali/kiali GitHub repository].
+
+This is the component that communicates with Istio parts, retrieves and
+processes data, and send processed data to the frontend.
+
+The backend doesn't need storage. When running the backend locally, all
+configurations are set in files and environment variables (including the user
+and password used for authentication). When deploying the backend to a cluster
+using the docker images, configmaps and secrets are used to set the
+configurations.
+
+== Kiali frontend
+
+The frontend is a single page web application, built with React and written in
+Typescript. The code can be found at
+https://github.com/kiali/kiali-ui[kiali/kiali-ui GitHub repository].
+
+In a standard deployment, the backend serves the frontend. Then, the frontend
+queries the Kiali backend in order to get data and present it to the user. 
+
+Since, currently, there are no options for personalizations, it's mainly
+stateless. Some data may be persisted - like session credentials -, but this
+data is stored in the browser and won't be available in other browsers nor
+other devices.
+
+== Istio
+
+Istio is a Kiali requirement. It's the component that provides and controls the
+service mesh. Although Kiali and Istio can be installed separately, Kiali
+depends on Istio and won't work if it's not present.
+
+Kiali needs to retrieve Istio data and configurations, which are exposed
+through Prometheus and the cluster API. This is the reason the diagram shows a
+dashed line: to denote indirect dependency.
+
+== Prometheus
+
+Prometheus is an Istio dependency. When Istio telemetry is enabled, metrics
+data is stored in Prometheus. Kiali uses the data stored in Prometheus to
+figure out the mesh topology, show metrics, calculate health, show possible
+problems, etc.
+
+Kiali communicates directly with Prometheus. This effectively turns Prometheus
```

Comment:
`Kiali communicates directly with Prometheus, assumes the data schema used by Istio Telemetery, and as such, is a hard dependency for Kiali, and most of it's features will not work without it.`

Reply:
I used your suggestion, changing it a little.
But I'm unsure, I get confused. It's unclear if Prometheus is the hard dependency, or if Istio is the hard dependency.